diff --git a/dist/danmaku.canvas.js b/dist/danmaku.canvas.js
index 7ce2902bc44f659716e3272f7774d000df11fd7b..d26524a66789f6754c1dd88e0a2aa0cbe9e28227 100644
--- a/dist/danmaku.canvas.js
+++ b/dist/danmaku.canvas.js
@@ -283,9 +283,9 @@
     (
       typeof window !== 'undefined' &&
       (
-        window.requestAnimationFrame ||
-        window.mozRequestAnimationFrame ||
-        window.webkitRequestAnimationFrame
+        window.requestAnimationFrame.bind(window) ||
+        window.mozRequestAnimationFrame.bind(window) ||
+        window.webkitRequestAnimationFrame.bind(window)
       )
     ) ||
     function(cb) {
diff --git a/dist/danmaku.js b/dist/danmaku.js
index aa05fb70adb2a39fe590b335515a227d1b8a263c..1d9580d9be718e55ccb3d75c33ac6e3a7f4dfa72 100644
--- a/dist/danmaku.js
+++ b/dist/danmaku.js
@@ -368,9 +368,9 @@
     (
       typeof window !== 'undefined' &&
       (
-        window.requestAnimationFrame ||
-        window.mozRequestAnimationFrame ||
-        window.webkitRequestAnimationFrame
+        window.requestAnimationFrame.bind(window) ||
+        window.mozRequestAnimationFrame.bind(window) ||
+        window.webkitRequestAnimationFrame.bind(window)
       )
     ) ||
     function(cb) {
@@ -381,9 +381,9 @@
     (
       typeof window !== 'undefined' &&
       (
-        window.cancelAnimationFrame ||
-        window.mozCancelAnimationFrame ||
-        window.webkitCancelAnimationFrame
+        window.cancelAnimationFrame.bind(window) ||
+        window.mozCancelAnimationFrame.bind(window) ||
+        window.webkitCancelAnimationFrame.bind(window)
       )
     ) ||
     clearTimeout;
diff --git a/dist/esm/danmaku.js b/dist/esm/danmaku.js
index 9b8a91877977113b39de202de95804c248f09fe4..d129e3d85c23521c3ac3eaae9fb8f99a811fdf84 100644
--- a/dist/esm/danmaku.js
+++ b/dist/esm/danmaku.js
@@ -362,9 +362,9 @@ var raf =
   (
     typeof window !== 'undefined' &&
     (
-      window.requestAnimationFrame ||
-      window.mozRequestAnimationFrame ||
-      window.webkitRequestAnimationFrame
+      window.requestAnimationFrame.bind(window) ||
+      window.mozRequestAnimationFrame.bind(window) ||
+      window.webkitRequestAnimationFrame.bind(window)
     )
   ) ||
   function(cb) {
@@ -375,9 +375,9 @@ var caf =
   (
     typeof window !== 'undefined' &&
     (
-      window.cancelAnimationFrame ||
-      window.mozCancelAnimationFrame ||
-      window.webkitCancelAnimationFrame
+      window.cancelAnimationFrame.bind(window) ||
+      window.mozCancelAnimationFrame.bind(window) ||
+      window.webkitCancelAnimationFrame.bind(window)
     )
   ) ||
   clearTimeout;
